### PR TITLE
Implemented Strategy design pattern with author ascending and descending

### PR DESCRIPTION
**Pull request by lbayliss-scottlogic**

### CODE REVIEW

File: src/test/java/com/scottlogic/AuthorPostSorterTest.java [hunk 1]
```diff
@@ -66,5 +66,5 @@ void sort_nullInput() {
         PostSorter authorPostSorter = new AuthorPostSorter();
         List<UserPost> nullInput = Arrays.asList();
         Assertions.assertEquals(nullInput, authorPostSorter.sort(null, SortOrder.ASC));
-    }
+    }*/
```
Comment (DKDiep) [inline src/test/java/com/scottlogic/AuthorPostSorterTest.java:69]: Bad practice to comment out test.
Use annotation (at)Ignore or (at)Disabled and add a comment to why this is disabled

File: src/main/java/com/scottlogic/Main.java [hunk 2]
```diff
@@ -8,29 +11,32 @@
...
     }
 
     /**
-     * Prints out all user posts
+     * Prints out user posts in Author-ascending order
      * @param userPosts The list of user posts to print out
      */
-    private static void printAllUserPosts(List<UserPost> userPosts) {
+    private static void sortAuthorAscending(List<UserPost> userPosts) {
+        SortStrategy sortAuthorAscending = new SortStrategy(new SortAuthorAscending());
```
Comment (DKDiep) [inline src/main/java/com/scottlogic/Main.java]: I find it odd you are instantiating your sort strategy in main and not your sorting class

File: src/main/java/com/scottlogic/SortStrategy.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package com.scottlogic;
+
+import java.util.List;
+
+public class SortStrategy {
```
Comment (DKDiep) [inline src/main/java/com/scottlogic/SortStrategy.java]: The strategy is usually the interface,

ie

Interface Sort
class BubbleSort implements SortStrategy
class InsertionSort implements SortStrategy

File: src/main/java/com/scottlogic/Main.java (hunk 2 above)
Comment (lbayliss-scottlogic) [reply src/main/java/com/scottlogic/Main.java]: How would I go about instantiating it into the sorting class?

**Pull request closed without merging**